Remove CFG_ from vexpress and various platforms

etienne-lms opened the pull request:
This P-R removes CFG_ definitions from platform_config.h files for the
platforms vexpress, dl02, hikey, marvell, mediatek, poplar, rcar, rpi3, sam,
sprd, zynqmp. 

For vexpress, there is a specific commit linaro can ack. All others aggregated 
linaro can ack the as obvious changes without asking maintainers.

jenswi-linaro commented on core/arch/arm/plat-hikey/conf.mk:
```diff
@@ -2,6 +2,9 @@ PLATFORM_FLAVOR ?= hikey
 include core/arch/arm/cpu/cortex-armv8-0.mk
 
+CFG_TEE_CORE_NB_CORE = 8
+
```
Empty line not needed.

etienne-lms replied:
addressed. remaining empty line is ok.

jenswi-linaro commented on core/arch/arm/plat-rpi3/conf.mk:
```diff
@@ -1,5 +1,8 @@
 include core/arch/arm/cpu/cortex-armv8-0.mk
 
+CFG_TEE_CORE_NB_CORE = 4
+
```
Empty line not needed.

etienne-lms replied:
addressed. remaining empty line is ok.

jenswi-linaro commented:
`"plat-vexpress: move CFG_TEE_CORE_NB_CORE to platform conf.mk"`
`Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>`

etienne-lms commented:
tags applied, rebased.

jforissier commented:
`plat-xxx` does not exist

etienne-lms commented:
True. I found this header line more easy track related to those for the other platforms.
Would you prefer `move CFG_TEE_CORE_NB_CORE to conf.mk for various platforms`?

etienne-lms commented:
Rebased and updated 2nd commit header line from
`plat-xxx: move CFG_TEE_CORE_NB_CORE from platform_config to conf.mk`
to
`core: move CFG_TEE_CORE_NB_CORE to conf.mk for various platforms`.
